### PR TITLE
Multithreaded range proof checks

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -992,6 +994,153 @@ bool CheckTransaction(const CTransaction& tx, CValidationState &state)
     return true;
 }
 
+extern secp256k1_context* secp256k1_bitcoin_verify_context;
```
Comment (jtimon) [inline src/main.cpp]: Nit: would like this in consensus/* rather than in main.cpp despite depneding on CCoinsViewCache/coins.

**Comment by jtimon** [comment]
fast-read ack

**Comment by jtimon** [comment]
Apart from my general dislike on adding more code to main.cpp (making things harder to read, rebase, blame, etc), it would be also nice that VerifyAmounts took a CValidationState as parameter to return different error strings in different cases.

**Comment by maaku** [comment]
Merged in #143 .

**Comment by instagibbs** [comment]
Included in `elements-0.13.1`, closing

**Pull request closed without merging**